lib/MusicBrainz/Server/Edit/Relationship/Edit.pm

```diff
@@ -240,6 +241,10 @@ sub build_display_data {
     my $old = $self->data->{old};
     my $new = $self->data->{new};
 
+    $self->c->model('LinkAttributeType')->load(map {
+        $_->all_attributes
+    } values %{ $loaded->{LinkType} });
+
```

Comment:
I was confused by this too. I found an edit that failed to load without it, so it might have been an unrelated, existing bug. The issue was that linkPhrase.js needed access to all the attributes associated with the link type, IIRC. I'll have to find what edit that was.

Reply:
Figured this out. The line in `Entity::LinkType` where we do `$self->link_entity('link_attribute_type', $_->type_id, $_->type);` was storing `undef` for those link attribute types, because they weren't loaded on the link type. And my commit titled "Further avoid infinite recursion in add_linked_entity" broke things because I changed the `!defined` check to `!exists`, so the defined ones from the relationship weren't added due to the `undef` values stored from the link type earlier.

Fixed it by changing that commit to check if `$entity` is defined before storing it. Reverted this.